src/plugins/sync-plugin.js

```diff
@@ -757,7 +822,13 @@ export const updateYFragment = (y, yDomFragment, pNode, mapping) => {
       const leftP = pChildren[left]
       const rightY = yChildren[yChildCnt - right - 1]
       const rightP = pChildren[pChildCnt - right - 1]
-      if (leftY instanceof Y.XmlText && leftP instanceof Array) {
+      if (!(leftP instanceof Array) && isFallbackNode && isFallbackNode(leftP)) {
+        if (mapping.get(leftY) !== leftP) {
+          yDomFragment.delete(left, 1)
+          yDomFragment.insert(left, /** @type Array{Y.XmlFragment} */ [leftY])
```

Comment:
I don't really see why it is necessary to delete and insert the same node. Also, it is not possible to insert the same shared type (`leftY`) again in the document. Even if it has been deleted before.